### PR TITLE
Add Kotlin integration and try to fix Hearthbleed

### PR DESCRIPTION
**Pull request by flibustier**
Transform from Java to Kotlin :
RSA is now an object
PublicKey and PrivateKey are now classes (in RSA)
Keys in Sender are now stored as Pair<PublicKey, PrivateKey> (thanks to Kotlin)
And lots of optimization for code readability (like using operators for BigInteger, …)

### CODE REVIEW

File: src/Main.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Created by John.
+ */
+object Main {
+    @JvmStatic fun main(args: Array<String>) {
+        val bob: Sender?
+
+        try {
+            bob = Sender("Bob", BigInteger("3"))
+        } catch (e: Exception) {
```
Comment (Slashgear) [inline src/Main.kt]: Catch Exception ?

File: src/Main.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            System.err.println(e.message)
+            return
```
Comment (Slashgear) [inline src/Main.kt]: return ?

File: src/Main.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        val eve = Hacker()
+
+        val plaintext = BigInteger("1337")
+        println("Plaintext > " + plaintext)
+
+        val cypher1 = RSA.encrypt(plaintext, bob.publicKey)
+        println("Cypher > " + cypher1)
+
+        System.out.println("Result > " + bob.decrypt(cypher1))
```
Comment (Slashgear) [inline src/Main.kt]: System.out

File: src/Main.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        val plaintext2 = BigInteger("1338")
+        println("Plaintext > " + plaintext2)
+
+        val cypher2 = RSA.encrypt(plaintext2, bob.publicKey)
+        println("Cypher > " + cypher2)
+
+        System.out.println("Result > " + bob.decrypt(cypher2))
```
Comment (Slashgear) [inline src/Main.kt]: idem

File: src/RSA.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            e = e.nextProbablePrime()
+        }
+
+        return keygen(e)
+    }
+
+    @Throws(Exception::class)
+    fun keygen(e: BigInteger): Pair<PublicKey, PrivateKey> {
+        if (!primalTestFermat(e))
+            throw Exception("e must be primal!")
```
Comment (Slashgear) [inline src/RSA.kt]: Create your own exception

File: src/Sender.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Created by John.
+ */
+
+import java.math.BigInteger
+
+class Sender {
+
+    private var keys: Pair<RSA.PublicKey, RSA.PrivateKey>
+    val name: String
```
Comment (Slashgear) [inline src/Sender.kt]: Two class with same attribut 'name' ?

File: src/Main.java [hunk 7]
```diff
@@ -1,35 +0,0 @@
...
-            return;
-        }
-
-        Hacker eve = new Hacker();
-
-        BigInteger plaintext = new BigInteger("1337");
-        System.out.println("Plaintext > " + plaintext);
-
-        BigInteger cypher1 = RSA.encrypt(plaintext ,bob.getPublicKey());
-        System.out.println("Cypher > " + cypher1);
```
Comment (Slashgear) [inline src/Main.java:20]: "Cypher > $cypher1", idem for other strings

File: src/Main.java [hunk 8]
```diff
@@ -1,35 +0,0 @@
-import java.math.BigInteger;
-
-public class Main {
-    public static void main(String[] args) {
```
Comment (Slashgear) [inline src/Main.java:4]: Where are tests ?

**Review by Slashgear — CHANGES_REQUESTED** [review]

File: src/Main.java (hunk 8 above)
Comment (flibustier) [reply src/Main.java:4]: Need help to initialize Spek tests ;)

**Pull request merged**